forgerock-authenticator/src/main/java/org/forgerock/android/authenticator/PushParser.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2020 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.authenticator;
+
+import org.forgerock.android.authenticator.exception.URIMappingException;
+import org.forgerock.util.encode.Base64;
+import org.forgerock.util.encode.Base64url;
+
+import java.util.Map;
+
+/**
+ * Provides the ability to parse URI scheme into a convenient format
+ * to use with configuring a {@link Push} to receive push notifications.
+ */
+public class PushParser extends UriParser {
```

Comment:
Never use. Do we really need this class?

Reply:
Yes, it was design to parse OTP and Push the URLs in their respective models objects. The same approach is used in iOS. But iOS is accepting the URL in the constructor on the model class. I was thinking in a different approach to Android. But I will align it to iOS SDK.